core/fetcher.go

```diff
@@ -33,11 +32,12 @@ var (
 )
 
 type BlockFetcher struct {
+	ctx    context.Context
```

Comment:
Nit: Don't need to store ctx, it is anti-pattern. It can be passed to all routines from Start call